File: ios/App/App/API/DirectoryFrontendClient.swift
// swift-format-ignore-file
//
//  DirectoryFrontendClient.swift
//  App
//
//  Copyright © 2020 MemeLabs. All rights reserved.
//

import Foundation
import PromiseKit

class DirectoryFrontendClient {
  private var client: RPCClient

  init(_ client: RPCClient) {
    self.client = client
  }

  public func open(_ arg: PBDirectoryFrontendOpenRequest = PBDirectoryFrontendOpenRequest()) throws -> RPCResponseStream<PBDirectoryFrontendOpenResponse> {
    return try self.client.callStreaming("DirectoryFrontend/Open", arg)
  }
  public func test(_ arg: PBDirectoryFrontendTestRequest = PBDirectoryFrontendTestRequest()) -> Promise<PBDirectoryFrontendTestResponse> {
    return self.client.callUnary("DirectoryFrontend/Test", arg)
  }
  
}

File: android/app/src/main/java/gg/strims/ppspp/rpc/DirectoryFrontendClient.kt
package gg.strims.ppspp.rpc
import gg.strims.ppspp.proto.*

class DirectoryFrontendClient(filepath: String) : RPCClient(filepath) {

    suspend fun open(
        arg: DirectoryFrontendOpenRequest = DirectoryFrontendOpenRequest()
    ): RPCResponseStream<DirectoryFrontendOpenResponse> =
        this.callStreaming("DirectoryFrontend/Open", arg)

    suspend fun test(
        arg: DirectoryFrontendTestRequest = DirectoryFrontendTestRequest()
    ): DirectoryFrontendTestResponse =
        this.callUnary("DirectoryFrontend/Test", arg)

}
